src/main/scala/sigmastate/Values.scala

```diff
@@ -251,9 +273,9 @@ object Values {
 
   implicit class CollectionConstantOps[T <: SType](c: CollectionConstant[T]) {
     def toConcreteCollection: ConcreteCollection[T] = {
-      implicit val tElem = c.tpe.elemType
+      val tElem = c.tpe.elemType
       val items = c.value.map(v => tElem.mkConstant(v.asInstanceOf[tElem.WrappedType]))
-      ConcreteCollection(items)
+      mkConcreteCollection(items, tElem).asInstanceOf[ConcreteCollection[T]]
```

Comment:
@greenhat I don't think we should use builder here, because this helper method works as typecast operation.
But let's leave it as it is.